src/core/verify.rs

```diff
@@ -205,6 +212,88 @@ pub const VERIFY_ALL_PRE_TAPROOT: btck_ScriptVerificationFlags = VERIFY_P2SH
     | VERIFY_CHECKSEQUENCEVERIFY
     | VERIFY_WITNESS;
 
+/// Precomputed transaction data for verifying a transasction's scripts.
+///
+/// Precomputes the hashes required to verify a transaction and avoids quadratic
+/// hashing costs when verifying multiple scripts from a transaction.
+///
+/// PrecomputedTransactionData is created from a transaction, and if doing
+/// taproot verification, its previous outputs [`crate::TxOut`]. It is required
+/// to perform script verification.
+///
+/// Previous outputs are only required if verifying a taproot transaction. An
+/// empty slice may be passed in otherwise.
+///
+/// # Examples
+///
+/// Creating a PrecomputedTransactionData:
+///
+/// # Returns
+///
+/// * `Ok(...)` - The PrecomputedTransactionData
+/// * `Err(KernelError::MismatchedOutputsSize)` - Number of outputs does not match
+/// the number of the transaction's inputs.
+///
+/// ```no_run
+/// # use bitcoinkernel::{prelude::*, Transaction, TxOut, PrecomputedTransactionData};
+/// # let raw_tx = vec![0u8; 100]; // placeholder
+/// # let tx = Transaction::new(&raw_tx).unwrap();
+/// # let tx_data = PrecomputedTransactionData::new(&tx, &Vec::<TxOut>::new());
```

Comment:
Typo? Looks like `# Examples` and `# Returns` are out-of-order.

Reply:
Thanks, fixed :)